server/src/routes/runRoutes.ts

```diff
@@ -257,128 +88,42 @@ export default function (
-                                    state: 'success',
-                                    progress: 100
-                                }
-                            }
-                        }
-                    }
-                });
+                if (contentId) {
+                    await fs.unlink(filePath);
+                }
+                res.status(200).json(response.body);
             } catch (error) {
-                Sentry.captureException(error);
-                return websocket.emit('action', {
-                    type: 'action',
-                    payload: {
-                        type: 'RUN_UPDATE_STATE',
-                        payload: {
-                            showDialog: true,
-                            uploadProgress: {
-                                import: {
-                                    state: 'success'
-                                },
-                                export: {
-                                    state: 'success'
-                                },
-                                upload: {
-                                    state: 'error',
-                                    progress: 0
-                                }
-                            }
-                        }
-                    }
-                });
+                res.status(error.status || 500).json(
+                    new LumiError(
+                        error.code,
+                        error.message,
+                        error.status,
+                        error
+                    )
+                );
             }
-
-            res.status(200).json(run);
         }
     );
 
-    router.delete(
-        '/:id',
-        async (
-            req: express.Request,
-            res: express.Response,
-            next: express.NextFunction
-        ) => {
-            try {
-                const response = await superagent.delete(
-                    `http://lumi.run/${req.params.id}?secret=${req.query.secret}`
-                );
-
-                const run = await fs.readJSON(serverConfig.runFile);
-                run.runs = run.runs.filter((r) => r.id !== req.params.id);
-                await fs.writeJSON(serverConfig.runFile, run);
-
-                res.status(200).json(response);
-            } catch (error) {
-                if (error.status === 404) {
-                    const run = await fs.readJSON(serverConfig.runFile);
-                    run.runs = run.runs.filter((r) => r.id !== req.params.id);
-                    await fs.writeJSON(serverConfig.runFile, run);
-                }
-                Sentry.captureException(error);
-                res.status(500).end();
+    router.use(
+        '/',
+        (req, res, next) => {
+            if (settingsCache.getSettings().token) {
+                req.headers['x-auth'] = settingsCache.getSettings().token;
             }
-        }
+            next();
+        },
+        proxy(runHost)
```

Comment:
I still don't believe it's a good idea to proxy all requests like this. It might make it difficult to add a forward proxy. Why not simply change the client to directly reach out to the server?